ldap_jwt_auth/core/schemas.py

```diff
@@ -14,3 +15,20 @@ class UserCredentialsPostRequestSchema(BaseModel):
     password: SecretStr
 
     model_config = ConfigDict(hide_input_in_errors=True)
+
+
+class MaintenanceState(BaseModel):
+    """
+    Model for maintenancce response
+    """
+
+    show: bool
+    message: str
+
+
+class ScheduledMaintenanceState(MaintenanceState):
+    """
+    Model for scheduled maintenance state
+    """
+
+    severity: Optional[str]
```

Comment:
This is SciGateway specific and I don't think the app should deal with stuff like this.

Reply:
In that case should I remove it from the json files etc. It may be scigateway specific but surely they would need to be able to edit it?

Reply:
[scigateway-auth](https://github.com/ral-facilities/scigateway-auth/blob/main/maintenance/scheduled_maintenance.json) does not store/return a `severity` so there must be something in SciGateway that decides on the value for it. Do you know what the possible `severity` values are?

Reply:
>[scigateway-auth](https://github.com/ral-facilities/scigateway-auth/blob/main/maintenance/scheduled_maintenance.json) does not store/return a `severity` value so there must be something in SciGateway that decides on the value for it.

I think this might be it https://github.com/ral-facilities/scigateway-auth/blob/main/maintenance/scheduled_maintenance.json. Maybe @joelvdavies has a better idea?

> Do you know what the possible `severity` values are?

Looks like the values are `'success' | 'warning' | 'error' | 'information'` so `warning` should be fine.

Reply:
Looking at it again, yes it is allowed to be undefined in which case it defaults to warning on SciGateway - see https://github.com/ral-facilities/scigateway/blob/a20de6eaaddb78c1e38d34f33d5e8a4c427cebb4/src/state/actions/scigateway.actions.tsx#L443

Reply:
https://github.com/ral-facilities/scigateway/blob/a20de6eaaddb78c1e38d34f33d5e8a4c427cebb4/src/state/actions/scigateway.actions.tsx#L460
. This is what Joel linked in a comment

Reply:
Thank you both. So in conclusion, you can remove the `severity` field from the `ScheduledMaintenanceState` class (but still keep the class) as SciGateway will take care of the `severity` value.